lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3FileSystemStats.java

```diff
@@ -0,0 +1,222 @@
+                            }
+
+                            if (value.equals(Boolean.FALSE)) {
+                                increment(stats.total.getFailures());
+                                if (apiCallStats != null) {
+                                    increment(apiCallStats.getFailures());
+                                }
+                            }
+                        }
+                        else if (metricRecord.metric().equals(RETRY_COUNT)) {
+                            Integer value = (Integer) metricRecord.value();
+
+                            stats.total.getRetries().update(value);
+                            if (apiCallStats != null) {
+                                apiCallStats.getRetries().update(value);
+                            }
+                        }
+                        else if (metricRecord.metric().equals(API_CALL_DURATION)) {
+                            Duration value = (Duration) metricRecord.value();
+
+                            stats.total.getLatency().addNanos(value.toNanos());
+                            if (apiCallStats != null) {
+                                apiCallStats.getLatency().addNanos(value.toNanos());
+                            }
+                        }
+                        else if (metricRecord.metric().equals(ERROR_TYPE)) {
+                            String value = (String) metricRecord.value();
+
+                            if (value.equals(THROTTLING.toString())) {
+                                increment(stats.total.getThrottlingExceptions());
+                                if (apiCallStats != null) {
+                                    increment(apiCallStats.getThrottlingExceptions());
+                                }
+                            }
+                            else if (value.equals(SERVER_ERROR.toString())) {
+                                increment(stats.total.getServerErrors());
+                                if (apiCallStats != null) {
+                                    increment(apiCallStats.getServerErrors());
+                                }
+                            }
+                        }
+                    });
+
+            metricCollection.children().forEach(child -> publishMetrics(child, apiCallStats));
+        }
+
+        public void close()
+        {
+        }
+
+        private AwsSdkV2ApiCallStats getApiCallStats(String operationName)
+        {
+            return switch (operationName) {
+                case "GetObject" -> stats.getObject;
+                case "ListObjectsV2Paginator" -> stats.listObjectsV2Paginator;
+                case "PutObject" -> stats.putObject;
+                case "CreateMultipartUpload" -> stats.createMultipartUpload;
+                case "CompleteMultipartUpload" -> stats.completeMultipartUpload;
+                case "AbortMultipartUpload" -> stats.abortMultipartUpload;
+                default -> null;
```

Comment:
Which API calls are not handled here?

Reply:
I added headObject, deleteObject, deleteObjects, uploadPart apis in the revision. This should cover all the apis which are currently being used in the native s3 filesystem.